src/App.tsx

```diff
@@ -1,14 +1,68 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+import { getTodos, getUser } from './api';
+
+enum Status {
+  ALL = 'all',
+  COMPLETED = 'completed',
+  ACTIVE = 'active',
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [loadingTodo, setLoadingTodo] = useState(true);
+
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [loadingUser, setLoadingUser] = useState(true);
+
+  const [selectedStatus, setSelectedStatus] = useState<Status>(Status.ALL);
+
+  const [query, setQuery] = useState('');
+
+  const handleSelectedStatus = (
+    event: React.ChangeEvent<HTMLSelectElement>,
+  ) => {
+    setSelectedStatus(event.target.value as Status);
```

Comment:
Casting `event.target.value` to `Status` without validation can lead to runtime errors if the value is not a valid `Status` enum. Consider validating the value before setting the state.